docs/adr/0001-Logging-warnings-using-the-Python-warnings-or-logging-module.md

```diff
@@ -0,0 +1,33 @@
+# 1. Logging warnings using the python warnings or logging module
+
+Date: 2022-10-14
+
+## Status
+
+Accepted
+
+## Context
+
+Python has two different ways to report warnings to the user. Warnings can be logged via the logger (`LOGGER.warning()`) or using the Python warnings module (`warnings.warn()`). It was unclear to developers which type of logger has to used in which circumstances.
+
+We don't want Deprecation warnings in third-party libraries to be visible to end-users.
+
+## Pros and Cons
+
+* LOGGER.warning():
+   * This logger just writes a message to the log.
+   * When the same message is logged twice, it will appear twice in the log.
+* warnings.warn():
+   * Each message is displayed only once.
+   * Pytest provides a nice overview of all warning logged during the executing of the test suite.
+
+## Decision
+
+- The `warnings` module will be configured as such that all warnings not extending from the `warnings.InmantaWarning` class, are not displayed to the user. This ensures that warnings from third-party libraries are always ignored.
+- All warnings logged to `warning.warn()` should extend from the `warnings.InmantaWarning` class. The `inmanta.warnings.warn()` is a wrapper around `warnigs.warn()` that ensures that this constraint is satisfied. Using the latter method is preferred.
+
+- All warnings, for which we expect an action from the end-user, should be logged using the `inmanta.warnings.warn()` method. An example is a log message that announces a feature deprecation.
+- All other warnings should be logged using the `LOGGER.warning()` method.
```

Comment:
I'm not following with the motivation for this yet. Revived the slack discussion.

Reply:
Could you elaborate on what you would like to see changed here?

Reply:
I'm honestly not sure what exactly but it feels a bit ambiguous to me. Bart and I are discussing it on [Slack](https://inmanta.slack.com/archives/CKRF0C8R3/p1666013643259899?thread_ts=1665730774.107619&cid=CKRF0C8R3), so perhaps let's centralize the discussion there.